src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,17 +1,121 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.IBookRepository;
 
-@Controller
+
+/**
+ * This class is the service responsible of handling the requests related to the book api
+ */
+
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private IBookRepository bookRepository;
+
     @GetMapping("/greeting")
-    public String greeting(@RequestParam(name="name", required=false, defaultValue="Poli") String name, Model model) {
+    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "Poli") String name,
+            Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
 
+    /**
+     * @return all the books contained in the model
+     */
+    @GetMapping("/all")
+    public Optional<List<Book>> findAll() {
+        List<Book> allBooks = new ArrayList<>();
+        if (bookRepository.findAll().iterator().hasNext()) {
+            bookRepository.findAll().forEach(b -> allBooks.add(b));
+            return Optional.of(allBooks);
+        } else {
+            return Optional.empty();
+        }
```

Comment:
This is a tip: `Optionals`are better to use with a single object. Container types, including collections, maps, streams, arrays, and optionals itself should not be wrapped in optionals. On the other hand, you can return the list from the repository, if there are no items in the database, it returns an empty list.
```suggestion
    public Optional<List<Book>> findAll() {
        return bookRepository.findAll();
```